### ISSUE TITLE
Website missing accessibility features for keyboard navigation

### ISSUE THREAD

**Issue by BoldBigflank** [trakt/trakt-lite#2201]
### Code of Conduct

- [x] I have read and agree to the project's [Code of Conduct](https://github.com/trakt/trakt-web/blob/main/.github/CODE_OF_CONDUCT.md)

### Where did you encounter the bug?

Throughout the site, aria roles are missing and other accessibility best practices are not followed.

### Please describe the bug below

Using a screen reader (I use NVDA), notice that the navigation section is not labeled as such, making it harder to return to it via key command.
In the search results screen, use tab to iterate through results. Notice that every result card is identified as "menu subMenu" and the link to the detail page is not selectable. Notice that there is no visual identification for what result is currently selected.

### Additional information

Testing as a developer:
Go to the main page. In developer tools, open Accessibility tab. Notice most nodes are labeled "generic" without any information. Notice that links are not included in the tab order.

**Comment by BoldBigflank** [comment]
Thanks @jasper-247 for creating an account and adding a comment to my Issue. It has nothing to do with the subject at hand, and reads like AI vomit.
